### PR TITLE
[ReviewMe] Analytics DB writing

### PR DESCRIPTION
**Pull request by jimrandomh**
See also matching PR against the credentials repo, which has the analytics DB connection string (without which this does nothing). The main entry point is `captureEvent` in `analyticsEvents.js`.

### CODE REVIEW

File: packages/lesswrong/server/analyticsWriter.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+if (!connectionString) {
+  //eslint-disable-next-line no-console
+  console.log("Analytics logging disabled: analytics.connectionString is not configured");
+}
+
+let analyticsConnectionPool = null;
+// Return the Analytics database connection pool, if configured. If no
+// analytics DB is specified in the server config, returns null instead. The
+// first time this is called, it will block briefly.
+export const getAnalyticsConnection = () => {
```
Comment (jimrandomh) [inline packages/lesswrong/server/analyticsWriter.js]:
```suggestion
const getAnalyticsConnection = () => {
```

Comment (Discordius) [inline packages/lesswrong/server/analyticsWriter.js]: Either also make into variable, or remove export

File: packages/vulcan-lib/lib/server/apollo-ssr/renderPage.js [hunk 2]
```diff
@@ -18,24 +20,74 @@ import AppGenerator from './components/AppGenerator';
...
     //eslint-disable-next-line no-console
     console.log(`Rendering ${req.url} (logged in request; hit rate=${getCacheHitRate()})`);
-    const rendered = await renderRequest(req, user);
-    sendToSink(sink, rendered);
+    const rendered = await renderRequest({
+      req, user, startTime
+    });
+    sendToSink(sink, {
+      ...rendered,
+      extraScripts: [tabIdHeader],
```
Comment (Discordius) [inline packages/vulcan-lib/lib/server/apollo-ssr/renderPage.js]: Probably call this variable `extraHeaders`

File: packages/vulcan-lib/lib/server/apollo-ssr/renderPage.js [hunk 3]
```diff
@@ -146,6 +201,9 @@ const sendToSink = (sink, {
   
   sink.appendToBody(ssrBody);
   sink.appendToHead(head);
+  if (extraScripts) {
```
Comment (Discordius) [inline packages/vulcan-lib/lib/server/apollo-ssr/renderPage.js]: Probably remove `extraScripts` and just transform `head` into a `heads` argument and have it take an array

File: packages/vulcan-lib/lib/server/apollo-ssr/renderPage.js [hunk 4]
```diff
@@ -181,16 +181,18 @@ const renderRequest = async ({req, user, startTime}) => {
...
     redirectUrl: serverRequestStatus.redirectUrl,
     timings,
   };
 }
 
 const sendToSink = (sink, {
-  ssrBody, head, serializedApolloState, jssSheets,
-  status, redirectUrl, extraScripts
+  ssrBody, headers, serializedApolloState, jssSheets,
+  status, redirectUrl, extraHeaders
```
Comment (Discordius) [inline packages/vulcan-lib/lib/server/apollo-ssr/renderPage.js]: extraHeaders is now unused

**Pull request merged**